### PR TITLE
Updating the empty, web and web api templates to beta4

### PR DESCRIPTION
**Pull request by aeschli**

### CODE REVIEW

**Comment by peterblazejewicz** [comment]
Guys, I think we also need to update Mocha tests. With this PR tests will fail. Some files are removed (so not created), some are changed.

**Comment by spboyer** [comment]
There are is a lot of discussion to be had concerning the templates for beta4. Tests, guplfile, xplat support removed in this PR. That's waht I've seen so far.

**Comment by peterblazejewicz** [comment]
To be strict: I meant our own files to test generator features:
https://github.com/OmniSharp/generator-aspnet/tree/master/test

File: templates/projects/web/_gulpfile.js [hunk 1]
```diff
@@ -1,19 +1,32 @@
...
+/// <binding Clean='clean' />
 
-gulp.task('default', ['bower:install'], function () {
-    return;
-});
+var gulp = require("gulp"),
+  rimraf = require("rimraf"),
+  fs = require("fs");
+
+eval("var project = " + fs.readFileSync("./project.json"));
```
Comment (peterblazejewicz) [inline templates/projects/web/_gulpfile.js]: Can this line be rewritten to not use eval? We should not ship eval to users in template as of 2015 I think:
```JavaScript
var project = require('./project.json');
var paths = {
  bower: "./bower_components/",
  lib: "./" + project.webroot + "/lib/"
};
```

Comment (spboyer) [inline templates/projects/web/_gulpfile.js]: That's what's in the 2015 gulp template. Part of my previous comment concerns.

Comment (sayedihashimi) [inline templates/projects/web/_gulpfile.js]: @spboyer I think I may have missed your feedback on the gulp template, could you share a link?

**Comment by sayedihashimi** [comment]
I think that we should create a new beta4 branch so that we can party on this together.
I'm planning to work on Console & Class Lib tmrw. Even though there are issues with this PR there is still a lot of value here. So I think we could use this as a starting point for the beta4 branch. Once we get issues ironed out and beta4 is released we can merge to master.

**Comment by aeschli** [comment]
I added 2 more commits to fix the tests and to add the kestrel command to all 3 projects

**Comment by spboyer** [comment]
@sayedihashimi agreed, I'll make a new branch "beta4" and then @aeschli can you do the PR against that branch so we can workout all the changes.

@peterblazejewicz

**Comment by spboyer** [comment]
@peterblazejewicz @sayedihashimi @aeschli 
beta4 branch is now available

**Comment by aeschli** [comment]
I created https://github.com/OmniSharp/generator-aspnet/pull/134 as pull request against the new beta4 branch

**Comment by peterblazejewicz** [comment]
Thanks guys! :clap: I was busy at work.

**Pull request closed without merging**